### PR TITLE
Resource interface : Cleanup and Quota

### PR DESCRIPTION
**Pull request by ihsaan-ullah**
# @ mention of reviewers
@Didayolo 

# A brief description of the purpose of the changes contained in this PR.
This PR solves the following points

- [x] show remaining quota (done but hidden til functionality is implementad by Tristan)
- [x] show `unused tasks` and delete option
- [x] show `unused datasets and programs` and delete option
- [x] show `unused submissions` and delete option
- [x] show `failed submissions` and delete option
- [x] implement delete api for `unused tasks`
- [x] implement delete api for `unused datasets and programs`
- [x] implement delete api for `unused submissions`
- [x] implement delete api for `unused failed submissions`
- [x] refresh resources after cleanup
- [x] refresh quota and cleanup after adding new submission/dataset/task
- [x] refresh quota and cleanup after deleting single submission/dataset/task  
- [x] refresh quota and cleanup after deleting multiple submission/dataset/task  

# Issues this PR resolves
#713 -> Interface -> Points: 12,  13, 14, 15, 16

> UI for quota management and cleanup (A "progress bar" for the storage quota)
> Delete unused tasks
> Delete unused datasets and programs
> Delete unused submissions
> Delete failed submissions

# Screenshots
with quota progress bar (hidden now)
244136599-77042dca-74fe-4f47-8b19-e84452126bd0

Working UI based on new api for getting cleaning statistics 
Screenshot 2023-06-08 at 1 49 20 AM

# Checklist
- [x] Code review by me 
- [x] Hand tested by me 
- [x] I'm proud of my work
- [x] Code review by reviewer
- [x] Hand tested by reviewer
- [x] CircleCi tests are passing
- [x] Ready to merge

### CODE REVIEW

**Comment by ihsaan-ullah** [comment]
To discuss 

- the difference between `Submission` and `Data` with type *submission*
- failed `Submission` when deleted still remains as `Data`

**Comment by Didayolo** [comment]
This looks very nice. I need some time to check this.

Just a quick question: what values did you use for the quota bar? It is just a fake quota or do you really compute some storage stuff here?

**Comment by Didayolo** [comment]
I rebased this pull request to avoid migration issues.

**Comment by Didayolo** [comment]
@ihsaan-ullah When I load the interface, I get the following error:

Capture d’écran 2023-06-13 à 12 21 45

**Comment by ihsaan-ullah** [comment]
I need more details. Error logs from docker django

**Comment by Didayolo** [comment]
Here are the Django logs:

```python
codabench-django-1  | 
codabench-django-1  | The above exception was the direct cause of the following exception:
codabench-django-1  | 
codabench-django-1  | Traceback (most recent call last):
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 34, in inner
codabench-django-1  |     response = get_response(request)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 115, in _get_response
codabench-django-1  |     response = self.process_exception_by_middleware(e, request)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 113, in _get_response
codabench-django-1  |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
codabench-django-1  |     return view_func(*args, **kwargs)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 71, in view
codabench-django-1  |     return self.dispatch(request, *args, **kwargs)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 495, in dispatch
codabench-django-1  |     response = self.handle_exception(exc)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 455, in handle_exception
codabench-django-1  |     self.raise_uncaught_exception(exc)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 492, in dispatch
codabench-django-1  |     response = handler(request, *args, **kwargs)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/rest_framework/decorators.py", line 54, in handler
codabench-django-1  |     return func(*args, **kwargs)
codabench-django-1  |   File "/app/src/apps/api/views/quota.py", line 19, in user_quota_cleanup
codabench-django-1  |     unused_tasks = Task.objects.filter(
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/models/query.py", line 392, in count
codabench-django-1  |     return self.query.get_count(using=self.db)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/models/sql/query.py", line 504, in get_count
codabench-django-1  |     number = obj.get_aggregation(using, ['__count'])['__count']
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/models/sql/query.py", line 489, in get_aggregation
codabench-django-1  |     result = compiler.execute_sql(SINGLE)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/models/sql/compiler.py", line 1142, in execute_sql
codabench-django-1  |     cursor.execute(sql, params)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 67, in execute
codabench-django-1  |     return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
codabench-django-1  |     return executor(sql, params, many, context)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
codabench-django-1  |     return self.cursor.execute(sql, params)
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/utils.py", line 89, in __exit__
codabench-django-1  |     raise dj_exc_value.with_traceback(traceback) from exc_value
codabench-django-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
codabench-django-1  |     return self.cursor.execute(sql, params)
codabench-django-1  | django.db.utils.ProgrammingError: more than one row returned by a subquery used as an expression
codabench-django-1  | 
```


Also, I think it would be cleaner to have this new interface at the bottom of the page, and with some (small) space between the two interfaces.

**Comment by ihsaan-ullah** [comment]
@Didayolo with the latest change, it should be working.

File: src/static/riot/quota_management.tag [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                            toastr.error(data.message)
+                        }
+                    })
+                    .fail(function (response) {
+                        toastr.error("Unsed tasks deletion failed!")
+                    })
+            }
+        }
+
+        // Delete unused tasks
```
Comment (Didayolo) [inline src/static/riot/quota_management.tag:119]: Should be // Delete unused datasets

File: src/static/riot/quota_management.tag [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                            toastr.error(data.message)
+                        }
+                    })
+                    .fail(function (response) {
+                        toastr.error("Unused datasets and programs deletion failed!")
+                    })
+            }
+        }
+
+        // Delete unused tasks
```
Comment (Didayolo) [inline src/static/riot/quota_management.tag:140]: Should be // Delete unused submissions

File: src/static/riot/quota_management.tag [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+                            toastr.error(data.message)
+                        }
+                    })
+                    .fail(function (response) {
+                        toastr.error("Unused submissions deletion failed!")
+                    })
+            }
+        }
+
+        // Delete unused tasks
```
Comment (Didayolo) [inline src/static/riot/quota_management.tag:161]: Should be // Delete failed submissions

**Comment by Didayolo** [comment]
I merge develop to have the last changes here.

**Pull request merged**